src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -102,7 +102,17 @@ public bool StartAssignment(HostAssignmentContext context)
         {
             if (!_webHostEnvironment.InStandbyMode)
             {
-                _logger.LogError("Assign called while host is not in placeholder mode");
+                // This is only true when specializing pinned containers.
+                if (!context.Environment.TryGetValue(EnvironmentSettingNames.ContainerStartContext, out string startContext))
```

Comment:
nit: for safety, add (|| string.IsNullEmpty(startContext)) in case something changes later on the platform side?

Reply:
I think we should continue failing in that case, because we should never be setting a null/empty env variable from the platform side.